docs/proposals/data-plane-config.md

```diff
@@ -0,0 +1,21 @@
+# Enhancement Proposal-929: Data Plane Dynamic Configuration
+
+- Issue: https://github.com/nginxinc/nginx-kubernetes-gateway/issues/929
+- Status: Provisional
+
+## Summary
+
+This proposal is intended to contain the design for how to dynamically configure the data plane for the
+NGINX Gateway Fabric (NGF) project. Similar to control plane configuration, we should be able to leverage
+a custom resource definition to define data plane configuration, considering fields such as telemetry and
+upstream zone size.
```

Comment:
I think we should configure upstream zone size dynamically.  
Configuring it globally go against memory efficiency. If we need a big zone for just a single upstream, but all other upstreams need smaller zones, this means all those other upstreams will waste memory.

Let's change it here to NGINX logging if we need an example and let's not commit to it in the zones.

Reply:
Do we see the value in allowing a global setting now, with the intention that future policy would allow for granular changes per upstream?

Reply:
Or does that conflict with automatic tuning that we may also plan for the future?

Reply:
Allowing a fixed value with a possibility to add an auto value in the future makes sense 👍 

A summary of an offline discussion:
- while the auto value will reduce memory, implementing it is not easy, and we have other much higher priorities right now
- having a knob is useful. if a user encounters a problem when the fixed zone cannot hold any more upstreams and thus NGINX will fail to reload, the knob will allow the user to increase the zone size and thus mitigate the issue.